src/libtorch.cc

```diff
@@ -819,13 +848,38 @@ ModelInstanceState::ValidateOutputs()
     std::string io_dtype;
     RETURN_IF_ERROR(io.MemberAsString("data_type", &io_dtype));
     const auto pr = ModelConfigDataTypeToTorchType(io_dtype);
-    if (!pr.first) {
+    if (!pr.first && (io_dtype != "TYPE_STRING")) {
       return TRITONSERVER_ErrorNew(
           TRITONSERVER_ERROR_INTERNAL,
           ("unsupported datatype " + io_dtype + " for output '" + io_name +
            "' for model '" + model_state_->Name() + "'")
               .c_str());
     }
+
+    // Validate shape for String outputs. Only allow 1 dimension and no
+    // batching.
+    if (io_dtype == "TYPE_STRING") {
+      // If a reshape is provided for the output then use that when
+      // validating the model shapes.
+      std::vector<int64_t> dims;
+      triton::common::TritonJson::Value reshape;
+      if (io.Find("reshape", &reshape)) {
+        RETURN_IF_ERROR(ParseShape(reshape, "shape", &dims));
+      } else {
+        RETURN_IF_ERROR(ParseShape(io, "dims", &dims));
+      }
+
+      if ((dims.size() + (supports_batching ? 1 : 0)) > 1) {
```

Comment:
Since this can't support batching at all, should be `if ((dims.size() > 1)  || supports_batching)`, otherwise the reshape case will pass the validation

Reply:
Done